Donation-platform/src/main/java/com/google/charities/CharityClone.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.charities;
+
+import java.util.Collection;
+
+/** Clone of a Charity to store data from charity JSONs: name, link, category, description, rating. */
+public final class CharityClone {
```

Comment:
What is the reason we can't use the default charity class?

Reply:
Because the purpose of this class is to exactly match the types in each json object in charities.json. This allows us to read in values from the json with one line: gson.fromJson(reader, CharityClone[].class);. If we used the Charity class then Gson wouldn't be able to match a string in the json object to a Tag or Cause object (for example) without further parsing. Essentially this class is to create objects that transfer charity attributes from the json into the new charity objects in the db.

Reply:
why can't we match the types in json with the existing class?

Reply:
I took a deeper look at Charity class and to me it seems feasible to directly populate using existing Charity Class. 

You can convert this json to JsonObject in java and then get the necessary fields from it to write it in db. We can discuss more in chat/gvc if you prefer.

Reply:
I am now using a HashMap. This allows for me to keep using the nice one-line conversion but without creating a whole new class to do so.